scripts/elasticsearch/test-elasticsearch.sh

```
@@ -0,0 +1,523 @@
+# Update the product and verify search reflects the change
+if [ -n "$SYNC_PRODUCT_ID" ]; then
+    UPDATED_NAME="SyncTest-Updated-Laptop-${TIMESTAMP}"
+    request PATCH "$PRODUCTSERVICE/products/$SYNC_PRODUCT_ID" "$AUTH_HEADERS" \
+        "{\"name\":\"${UPDATED_NAME}\"}"
+    assert_status "PATCH /products (update sync test product)" "200" "$STATUS"
+
+    sleep 3
+
+    # Old name should not return results
+    request GET "$PRODUCTSERVICE/search/products?query=${SYNC_PRODUCT_NAME}&page=0&size=10"
+    OLD_RESULTS=$(echo "$BODY" | python3 -c "import sys,json; print(json.load(sys.stdin).get('totalElements', 0))" 2>/dev/null || echo "0")
```

Shellcheck (0.11.0)

[warning] OLD_RESULTS appears unused. Verify use (or export if used externally).

(SC2034)